pkg/virt-operator/resource/apply/core.go

```diff
@@ -385,15 +385,15 @@ func (r *Reconciler) createOrUpdateRbac() error {
 
 	// create/update ClusterRoles
 	for _, cr := range r.targetStrategy.ClusterRoles() {
-		err := r.createOrUpdateClusterRole(cr, version, imageRegistry, id)
+		err := r.createOrUpdateRole(cr.DeepCopy(), version, imageRegistry, id, TypeClusterRole, false)
```

Comment:
I think this ends up causing more confusing and being harder to read. Especially combining Roles and RoleBindings into the same function, these are very different resources and the code for the create or update becomes very confusing between the two types I think in this case some code reuse is better. We currently in this package have an create/update function for each resource type this makes updates in the future easier if we need to change logic for that specific resource type.  I would lead towards keeping these as 4 separate functions for readability and future changes then as 2 generic functions or 1.

Reply:
Thanks for your feedback. I see your point and totally agree.

Since the implementation is nearly identical IMHO we should keep the 4 previous functions (createOrUpdateRole, createOrUpdateClusterRole, etc') but have them all call a private function that does the actual work. This way on the one hand we don't loose readability but on the other hand, if we need to change the implementation, we won't have to do so in 4 different places which are almost entirely duplicated.

What do you think?